### ISSUE TITLE
BE crash with space function

### ISSUE THREAD

**Issue by wangruin** [StarRocks/starrocks#5177]
### Steps to reproduce the behavior (Required)

1. CREATE EXTERNAL RESOURCE "hive_resource_emr" PROPERTIES ("type"="hive", "hive.metastore.uris"="thrift://172.26.194.238:9083")
2. CREATE EXTERNAL TABLE `hive_ssb_100g_orc_customer` (
  `c_custkey` int(11) NULL COMMENT "",
  `c_name` varchar(26) NULL COMMENT "",
  `c_address` varchar(41) NULL COMMENT "",
  `c_city` varchar(11) NULL COMMENT "",
  `c_nation` varchar(16) NULL COMMENT "",
  `c_region` varchar(13) NULL COMMENT "",
  `c_phone` varchar(16) NULL COMMENT "",
  `c_mktsegment` varchar(11) NULL COMMENT ""
) ENGINE=HIVE
COMMENT "PARTITION BY ()"
PROPERTIES (
"database" = "ssb_orc",
"table" = "customer",
"resource" = "hive_resource_emr",
"hive.metastore.uris"  =  "thrift://172.26.194.238:9083"
);
3. CREATE TABLE `tpch_100g_part` (
  `P_PARTKEY` int(11) NOT NULL COMMENT "",
  `P_NAME` varchar(55) NOT NULL COMMENT "",
  `P_MFGR` char(25) NOT NULL COMMENT "",
  `P_BRAND` char(10) NOT NULL COMMENT "",
  `P_TYPE` varchar(25) NOT NULL COMMENT "",
  `P_SIZE` int(11) NOT NULL COMMENT "",
  `P_CONTAINER` char(10) NOT NULL COMMENT "",
  `P_RETAILPRICE` decimal64(15, 2) NOT NULL COMMENT "",
  `P_COMMENT` varchar(23) NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`P_PARTKEY`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`P_PARTKEY`) BUCKETS 12
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
2. select   cast(     coalesce(       case         when ref_1.c_phone <= ref_1.c_mktsegment then ref_0.P_BRAND         else ref_0.P_BRAND       end,       ref_1.c_region     ) as VARCHAR   ) as c0,   ref_0.P_BRAND as c1,   ref_1.c_mktsegment as c2,   ref_1.c_city as c3,   ref_0.P_CONTAINER as c4,   cast(coalesce(28, ref_0.P_PARTKEY) as INT) as c5,   ref_0.P_CONTAINER as c6,   ref_0.P_TYPE as c7,   ref_1.c_region as c8,   ref_1.c_nation as c9,   ref_1.c_city as c10,   ref_0.P_TYPE as c11,   ref_0.P_NAME as c12,   cast(     nullif(       ref_0.P_SIZE,       case         when true then max(cast(ref_0.P_SIZE as INT)) over
(           partition by ref_1.c_address           order by             ref_0.P_BRAND,             ref_0.P_MFGR         )
    else max(cast(ref_0.P_SIZE as INT)) over (           partition by ref_1.c_address           order by             ref_0.P_BRAND,             ref_0.P_MFGR         )       end     ) as INT   ) as c13,   random() as c14 from   tpch_100g_part as ref_0
 right join hive_ssb_100g_orc_customer as ref_1 on (ref_0.P_BRAND = ref_1.c_name) where   (true)   and (     space(cast(ref_1.c_custkey as INT)) = cast(       nullif(ref_0.P_COMMENT, ref_1.c_address) as VARCHAR     )   ) limit   137;

### Expected behavior (Required)
Query returns correctly

### Real behavior (Required)
```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
*** Aborted at 1650114766 (unix time) try "date -d @1650114766" if you are using GNU date ***
PC: @     0x7f0430d1b337 __GI_raise
*** SIGABRT (@0x3e80000037e) received by PID 894 (TID 0x7f03d9d54700) from PID 894; stack trace: ***
    @     0x7f04317d05f0 (unknown)
    @     0x7f0430d1b337 __GI_raise
    @     0x7f0430d1ca28 __GI_abort
    @          0x1886271 _ZN9__gnu_cxx27__verbose_terminate_handlerEv.cold
    @          0x572a7e6 __cxxabiv1::__terminate()
    @          0x572a851 std::terminate()
    @          0x572a9a4 __cxa_throw
    @          0x1886178 _Znwm.cold
    @          0x344d992 std::vector<>::_M_fill_insert()
    @          0x344dd41 starrocks::vectorized::SpaceFunction::evaluate<>()
    @          0x344e0fb starrocks::vectorized::UnpackConstColumnUnaryFunction<>::evaluate<>()
    @          0x344e305 starrocks::vectorized::DealNullableColumnUnaryFunction<>::evaluate<>()
    @          0x3440c85 starrocks::vectorized::StringFunctions::space()
    @          0x310effc starrocks::vectorized::VectorizedFunctionCallExpr::evaluate()
    @          0x2c85afc starrocks::vectorized::VectorizedBinaryPredicate<>::evaluate()
    @          0x2bba1ac starrocks::ExprContext::evaluate()
    @          0x23bd5f3 starrocks::ExecNode::eval_conjuncts()
    @          0x2533306 starrocks::vectorized::HashJoinNode::_probe()
    @          0x2534251 starrocks::vectorized::HashJoinNode::get_next()
    @          0x267d3b1 starrocks::vectorized::ProjectNode::get_next()
    @          0x1f4a9c3 starrocks::PlanFragmentExecutor::_get_next_internal_vectorized()
    @          0x1f4c2ee starrocks::PlanFragmentExecutor::_open_internal_vectorized()
    @          0x1f4ced7 starrocks::PlanFragmentExecutor::open()
    @          0x1ee058b starrocks::FragmentExecState::execute()
    @          0x1ee4d2c starrocks::FragmentMgr::exec_actual()
    @          0x1ee5591 _ZNSt17_Function_handlerIFvvEZN9starrocks11FragmentMgr18exec_plan_fragmentERKNS1_23TExecPlanFragmentParamsERKSt8functionIFvPNS1_20PlanFragmentExecutorEEESC_EUlvE_E9_M_invokeERKSt9_Any_data
    @          0x20270a9 starrocks::ThreadPool::dispatch_thread()
    @          0x2022c68 starrocks::Thread::supervise_thread()
    @     0x7f04317c8e65 start_thread
    @     0x7f0430de388d __clone
    @                0x0 (unknown)
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
branch-2.2 release 2f51014

**Comment by wangruin** [comment]
select     case when 25 > unix_timestamp() then ref_1.P_MFGR else ref_1.P_MFGR end      as c0,    ref_0.s_address as c1,
 ref_1.P_BRAND as c2,    add_months(     cast(cast(null as DATETIME) as DATETIME),     cast(ref_1.P_PARTKEY as INT)) as c3,    ref_0.s_address as c4,    case when (ref_1.P_RETAILPRICE = cast(null as DOUBLE))        or ((ref_0.s_suppkey <= ref_0.s_suppkey)
     and (ref_1.P_BRAND >= ref_1.P_TYPE)) then 59 else 59 end      as c5,    ref_0.s_region as c6 from   ssb_100g_supplier as ref_0     inner join tpch_100g_part as ref_1     on (ref_0.s_name = ref_1.P_NAME ) where space(     cast(ref_1.P_PARTKEY as INT)) is not NULL limit 173;

```
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
*** Aborted at 1650783947 (unix time) try "date -d @1650783947" if you are using GNU date ***
PC: @     0x7f9985f09337 __GI_raise
*** SIGABRT (@0x3e80001cb33) received by PID 117555 (TID 0x7f995a7ab700) from PID 117555; stack trace: ***
    @     0x7f99869be5f0 (unknown)
    @     0x7f9985f09337 __GI_raise
    @     0x7f9985f0aa28 __GI_abort
    @          0x18872ed _ZN9__gnu_cxx27__verbose_terminate_handlerEv.cold
    @          0x572db26 __cxxabiv1::__terminate()
    @          0x572db91 std::terminate()
    @          0x572dce4 __cxa_throw
    @          0x18871f4 _Znwm.cold
    @          0x3451762 std::vector<>::_M_fill_insert()
    @          0x3451b49 starrocks::vectorized::SpaceFunction::evaluate<>()
    @          0x3451d83 starrocks::vectorized::UnpackConstColumnUnaryFunction<>::evaluate<>()
    @          0x3451f7b starrocks::vectorized::DealNullableColumnUnaryFunction<>::evaluate<>()
    @          0x34444b5 starrocks::vectorized::StringFunctions::space()
    @          0x311088c starrocks::vectorized::VectorizedFunctionCallExpr::evaluate()
    @          0x3174385 starrocks::vectorized::VectorizedIsNotNullPredicate::evaluate()
    @          0x2bbb968 starrocks::ExprContext::evaluate()
    @          0x1e3947c starrocks::vectorized::ColumnExprPredicate::evaluate()
    @          0x1af8c9c starrocks::vectorized::SegmentIterator::_filter_by_expr_predicates()
    @          0x1afa24a starrocks::vectorized::SegmentIterator::_do_get_next()
    @          0x1afd511 starrocks::vectorized::SegmentIterator::do_get_next()
    @          0x1b5b142 starrocks::vectorized::ProjectionIterator::do_get_next()
    @          0x1ea9e6a starrocks::SegmentIteratorWrapper::do_get_next()
    @          0x1b8559b starrocks::vectorized::TimedChunkIterator::do_get_next()
    @          0x1b7ed4e starrocks::vectorized::TabletReader::do_get_next()
    @          0x283437c starrocks::pipeline::OlapChunkSource::_read_chunk_from_storage()
    @          0x28349ee starrocks::pipeline::OlapChunkSource::buffer_next_batch_chunks_blocking()
    @          0x26cdc17 _ZNSt17_Function_handlerIFvvEZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS1_12RuntimeStateEiEUlvE0_E9_M_invokeERKSt9_Any_data
    @          0x1ed1110 starrocks::PriorityThreadPool::work_thread()
    @          0x3d4b187 thread_proxy
    @     0x7f99869b6e65 start_thread
    @     0x7f9985fd188d __clone
    @                0x0 (unknown)
```

branch-2.2 release 55c6918

**Comment by stdpain** [comment]
> starrocks::vectorized::ColumnExprPredicate::evaluate()

it was because function space has been pushed down to storage layer.

**Comment by kangkaisen** [comment]
> > starrocks::vectorized::ColumnExprPredicate::evaluate()
> 
> it was because function space has been pushed down to storage layer.

This issue only occur when enable_column_expr_predicate is true, so move this issue to 2.3 version.

**Comment by wangruin** [comment]
select * from   (     select       ref_0.ws_bill_cdemo_sk as c0     from       sqlsmith.tpcds_100g_web_sales as ref_0     where       ref_0.ws_sales_price = ref_0.ws_list_price   ) as subq_0 where   (     case       when subq_0.c0 > subq_0.c0 then space(cast(subq_0.c0 as INT))       else space(cast(subq_0.c0 as INT))     end >= case       when (         (false)         and (cast(null as VARCHAR) = cast(null as VARCHAR))       )       and (         (           (cast(null as DOUBLE) > cast(null as DOUBLE))           and (             (               (                 (subq_0.c0 is not NULL)                 and (                   (
 (subq_0.c0 is NULL)                     or (cast(null as VARCHAR) <= cast(null as VARCHAR))                   )                   or (subq_0.c0 <> subq_0.c0)
         )               )               or (                 (false)                 or (cast(null as DOUBLE) > cast(null as DOUBLE))               )             )
           and (subq_0.c0 is not NULL)           )         )         and (cast(null as DOUBLE) <= cast(null as DOUBLE))       ) then upper(         cast(           cast(             nullif(               case                 when subq_0.c0 is NULL then cast(null as VARCHAR)                 else cast(null as VARCHAR)
end,               case                 when (                   (cast(null as DOUBLE) = cast(null as DOUBLE))                   or (subq_0.c0 is NULL)
  )                 or (cast(null as DOUBLE) < cast(null as DOUBLE)) then cast(null as VARCHAR)                 else cast(null as VARCHAR)               end
   ) as VARCHAR           ) as VARCHAR         )       )       else upper(         cast(           cast(             nullif(               case                 when subq_0.c0 is NULL then cast(null as VARCHAR)                 else cast(null as VARCHAR)               end,               case                 when (
(cast(null as DOUBLE) = cast(null as DOUBLE))                   or (subq_0.c0 is NULL)                 )                 or (cast(null as DOUBLE) < cast(null as DOUBLE)) then cast(null as VARCHAR)                 else cast(null as VARCHAR)               end             ) as VARCHAR           ) as VARCHAR         )       )     end
  )   and (     cast(       nullif(         cast(nullif(subq_0.c0, subq_0.c0) as INT),         subq_0.c0       ) as INT     ) > subq_0.c0   );

```
  what():  std::bad_alloc
*** Aborted at 1651755085 (unix time) try "date -d @1651755085" if you are using GNU date ***
terminate called recursively
PC: @     0x7f4373290337 __GI_raise
*** SIGABRT (@0x3e8000061d8) received by PID 25048 (TID 0x7f4342ba1700) from PID 25048; stack trace: ***
    @          0x3c947d2 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7f4373d455f0 (unknown)
    @     0x7f4373290337 __GI_raise
    @     0x7f4373291a28 __GI_abort
    @          0x5653992 __gnu_cxx::__verbose_terminate_handler()
    @          0x5652446 __cxxabiv1::__terminate()
    @          0x56524b1 std::terminate()
    @          0x5652604 __cxa_throw
    @          0x17afa54 _Znwm.cold
    @          0x3384e52 std::vector<>::_M_fill_insert()
    @          0x3385239 starrocks::vectorized::SpaceFunction::evaluate<>()
    @          0x33856bd starrocks::vectorized::DealNullableColumnUnaryFunction<>::evaluate<>()
    @          0x3377ba5 starrocks::vectorized::StringFunctions::space()
    @          0x3042a5c starrocks::vectorized::VectorizedFunctionCallExpr::evaluate()
    @          0x2bb5e5c starrocks::vectorized::VectorizedBinaryPredicate<>::evaluate()
    @          0x2aed3e8 starrocks::ExprContext::evaluate()
    @          0x1d657fc starrocks::vectorized::ColumnExprPredicate::evaluate()
    @          0x1a24f5c starrocks::vectorized::SegmentIterator::_filter_by_expr_predicates()
    @          0x1a2650a starrocks::vectorized::SegmentIterator::_do_get_next()
    @          0x1a297d1 starrocks::vectorized::SegmentIterator::do_get_next()
    @          0x1a87502 starrocks::vectorized::ProjectionIterator::do_get_next()
    @          0x1dd621a starrocks::SegmentIteratorWrapper::do_get_next()
    @          0x1ab195b starrocks::vectorized::TimedChunkIterator::do_get_next()
    @          0x1ab72e4 starrocks::vectorized::UnionIterator::do_get_next()
    @          0x1aab10e starrocks::vectorized::TabletReader::do_get_next()
    @          0x2765d1c starrocks::pipeline::OlapChunkSource::_read_chunk_from_storage()
    @          0x276638e starrocks::pipeline::OlapChunkSource::buffer_next_batch_chunks_blocking()
    @          0x25ffc97 _ZNSt17_Function_handlerIFvvEZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS1_12RuntimeStateEiEUlvE0_E9_M_invokeERKSt9_Any_data
    @          0x1dfd510 starrocks::PriorityThreadPool::work_thread()
    @          0x3c2fe07 thread_proxy
    @     0x7f4373d3de65 start_thread
    @     0x7f437335888d __clone
```

**Comment by stdpain** [comment]
fixed by #6491
